### PR TITLE
base client: support list-releases

### PR DESCRIPTION
**Pull request by sergiusens**
Introduce the namespaced endpoint with pydantic validation.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
CRAFT-951

### CODE REVIEW

File: craft_store/base_client.py [hunk 1]
```diff
@@ -251,3 +251,10 @@ def progress_printer(monitor: requests_toolbelt.MultipartEncoderMonitor):
         logger.debug("Uploading bytes for %r ended, id %r", str(filepath), upload_id)
 
         return upload_id
+
+    def get_list_releases(self, *, name: str) -> models.MarshableModel:
+        """Query the list_releases endpoint and return the result."""
+        endpoint = f"/v1/{self._endpoints.namespace}/{name}/releases"
+        response = self.request("GET", self._base_url + endpoint).json()
+
+        return self._endpoints.get_list_releases.unmarshal(response)
```
Comment (maxiberta) [inline craft_store/base_client.py:260]: I initially thought `Endpoints.get_list_releases` was a function, but it is the expected response model (confusingly, it *is* a `BaseClient` method, too). I'd suggest maybe drop the `get` and add `model`? E.g. `list_releases_model`, `release_model`, `list_releases_item_model`, etc.

File: craft_store/endpoints.py [hunk 2]
```diff
@@ -30,20 +36,25 @@ class Package:
...
+    _param get_list_releases:
+        list_releases model response model for a given namespace.
     """
 
+    namespace: str
     whoami: str
     upload: str
     tokens: str
     tokens_exchange: str
+    get_list_releases: Type[MarshableModel]
```
Comment (maxiberta) [inline craft_store/endpoints.py:57]: Here, I think the `get_` prefix makes it look like a function.

File: craft_store/models/_base_model.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    class Config:  # type: ignore # pylint: disable=too-few-public-methods
+        """Pydantic model configuration."""
+
+        validate_assignment = True
+        allow_mutation = False
+        alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]) -> "MarshableModel":
+        """Create and populate a new ``PartSpec`` object from dictionary data.
```
Comment (maxiberta) [inline craft_store/models/_base_model.py:36]: What's `PartSpec` exactly?

**Review by maxiberta — APPROVED** [review]
Looks good! Just a couple of non blocking comments. Thanks

File: craft_store/base_client.py (hunk 1 above)
Comment (sergiusens) [reply craft_store/base_client.py]: this is a good observation!

**Review by cmatsuoka — APPROVED** [review]
Pydantic usage looks fine, all basic models without any advanced feature.

**Pull request merged**